### PR TITLE
Iteration4 filtering

### PR DESCRIPTION
**Pull request by orlov-n**
# Pull Request Template

## Description

Added filtering functionality.  Refactored CSS, HTML and javaScript to match Turing style guidelines.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Tested in the browser repeatedly.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [n/a] I have added tests that prove my fix is effective or that my feature works
- [n/a] New and existing unit tests pass locally with my changes
- [n/a] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] We had fun

### CODE REVIEW

**Comment by orlov-n** [comment]
The code was reviewed live by all Turing students via ZOOM.

File: main.js [hunk 1]
```diff
@@ -37,38 +44,102 @@ function hide(element) {
...
+        updateView(ideas);
+      }
+    }
+  }
+}
+
+function filterStarred(e) {
+  e.preventDefault();
+  if (buttonShowStarred.innerText === 'Show All Ideas') {
+    console.log('hi');
```
Comment (CHaiz15) [inline main.js]: Hi! I appreciate the welcoming but probably don't need this console log haha

File: main.js [hunk 2]
```diff
@@ -37,38 +44,102 @@ function hide(element) {
...
+    updateView(ideas);
+    buttonShowStarred.innerText = 'Show Starred Ideas';
+  } else {
+    var starredIdeas = ideas.map(function checkForStarred(element) {
```
Comment (CHaiz15) [inline main.js]: I like the effort to name but this could also be an anonymous function like 
```
.map((element) => blah
```

**Review by CHaiz15 — APPROVED** [review]

File: main.js (hunk 1 above)
Comment (zliibbe) [reply main.js]: console.log, removed from line 92

File: main.js (hunk 2 above)
Comment (zliibbe) [reply main.js]: Annonymous function in ES5 notation made on line 95 (not 96, due to earlier deleted console.log)

**Comment by zliibbe** [comment]
Code reviewed and merged live via Zoom with @c-o-de and @jskomal

**Pull request merged**